odoo/models.py

```diff
@@ -3235,30 +3221,7 @@ def fetch(self, field_names):
         if not self or not field_names:
             return
 
-        # determine fields to fetch
-        fields_to_verify = OrderedSet()
-        field_names = self.check_field_access_rights('read', field_names)
```

Comment:
We dropped that call in both `search_fetch()` and `fetch()`.
Either we reintroduce it in both methods, or we move it inside `_determine_fields_to_fetch()`.